src/drivers/uart/src/sunxi.rs

```diff
@@ -0,0 +1,170 @@
+use clock::ClockNode;
+use consts::DeviceCtl;
+use core::ops;
+use model::*;
+
+use tock_registers::interfaces::{ReadWriteable, Readable, Writeable};
+use tock_registers::register_bitfields;
+use tock_registers::registers::{ReadOnly, ReadWrite};
+
+const RETRY_COUNT: u32 = 100_000;
+
+/*
+ * see also https://github.com/smaeul/sun20i_d1_spl for reference
+ * files: nboot/main/boot0_head.c and drivers/serial.c
+ */
+
+// D1 user manual
+// https://dl.linux-sunxi.org/D1/D1_User_Manual_V0.1_Draft_Version.pdf
```

Comment:
if it is allowed, consider adding this to the oreboot datasheets repo.

Reply:
It has a watermark, says something about confidential etc, though anyone could also sign up for the Allwinner dev portal and download it. I'd rather ask Allwinner, though on the other hand, the linux-sunxi wiki has a bunch of resources on it that I'd not want to spoil. I have very mixed feelings.